doc/platform/common_config.md

```diff
@@ -0,0 +1,107 @@
+platform specific config. Duplicate configuration entries in the platform specific file override entries in the common config.bcm
+
+- The final config.bcm merged with common config is required to be copied to a shared folder for debugging   
+  
+
+## 2 Supported Platforms
+
+Per-switching silicon Common config feature is supported on all of the Broadcom platform if the common config files are created in the device/broadcom/x86_64-broadcom_common. Following is the current supported common configuration :
+
+```
+|-- x86_64-broadcom_b77 -- broadcom-sonic-td3.config.bcm
+|-- x86_64-broadcom_b85 -- broadcom-sonic-td2.config.bcm
+|-- x86_64-broadcom_b87-- broadcom-sonic-td3.config.bcm
+|-- x86_64-broadcom_b96-- broadcom-sonic-th.config.bcm
+|-- x86_64-broadcom_b97-- broadcom-sonic-th2.config.bcm
+|-- x86_64-broadcom_b98-- broadcom-sonic-th3.config.bcm
+```
+## 3 Design Detail
+The main change of the design is in the SYNCD docker syncd/scripts/syncd_init_common.sh script along with common config being created in the device/broadcom/x86_64-broadcom_common/ folder. The design standardize the common file name as broadcom-sonic-{$chip_id}.config.bcm . Also, in the SYNCD docker-syncd-brcm.mk, we extern the common config directory path /usr/share/sonic/device/x86_64-broadcom_common from host to docker for script reference.
+
+The design change for syncd_init_common.sh is targeted on the config_syncd_bcm() which is only for BRCM switch chip. In the config_syncd_bcm() function, it will load the common silicon config.bcm file by looking for a *.bcm file in the  /usr/share/sonic/device/x86_64-broadcom_common/broadcom-sonic-{$chip_id} folder where the $chip_id is the upper 3 nibbles of the switching silicon's device ID as listed in SDK/include/soc/devids.h. Then, merge common  broadcom-sonic-{$chip_id}.config.bcm  file with the existing platform specific config.bcm file in which the duplicate configuration entries in the platform specific file will override entries in the common broadcom-sonic-{$chip_id}.config.bcm file. 
+
+Since the platform specific config.bcm is read only in docker, the design copies the platform specific config.bcm and sai.profile to /tmp for handling common config merge process. The /tmp/sai.profile will be modified to point to the merged config.bcm under/tmp directory. The following switch initialization will reference to the new merged config.bcm pointed by updated sai.profile from the /tmp directory.
+
+
+## 4 Serviceability and DEBUG
+### 4.1 Syslogs
+During system booting syncd initialization, the Per-switching silicon common config design will log the common config merged information to the syslogs for identifying which common config be merged to the ODM config.bcm for syncd initialization.
+
+### Examples:
+```
+Nov  3 09:19:25.373964 2020 sonic INFO syncd#supervisord: syncd Merging 
+/usr/share/sonic/hwsku/td3-ix8a-bwde-48x25G+8x100G.config.bcm with
+/usr/share/sonic/device/x86_64-broadcom_common/x86_64-broadcom_b77
```

Comment:
What is the difference between the b77 and b87 version of the common td3 config.bcm?  from chip_id you can only tell if it is TD3/TH/TD2/... etc but how can you use that to select which one of the TD3 to use?  Can you elaborate on this?

Reply:
1. The b77 and b87 is different TD3 model which is the upper 3 nibbles of the switching silicon's device ID.

2. We use the CHIP ID to select the common config folder. FYI, see the mention in HLD :
In the config_syncd_bcm() function, it will load the common silicon config.bcm file by looking for a *.bcm file in the  /usr/share/sonic/device/x86_64-broadcom_common/broadcom-sonic-{$chip_id} folder where the $chip_id is the upper 3 nibbles of the switching silicon's device ID as listed in SDK/include/soc/devids.h.